examples/restore_login.py

```diff
@@ -1,71 +1,90 @@
+#!/usr/bin/env python3
+
 import asyncio
 import json
 import os
 import sys
+import getpass
 
 from nio import AsyncClient, LoginResponse
 
 CONFIG_FILE = "restore_login_config.json"
 
 # Check out main() below to see how it's done.
 
-def write_details_to_disk(resp: LoginResponse) -> None:
+
+def write_details_to_disk(resp: LoginResponse, home_server) -> None:
     """Writes the required login details to disk so we can log in later without
     using a password.
 
     Arguments:
         resp {LoginResponse} -- the successful client login response.
+        home_server -- the name of the homeserver, e.g. "https://matrix.example.org"
     """
     # open the config file in write-mode
     with open(CONFIG_FILE, "w") as f:
         # write the login details to disk
         json.dump(
             {
-            "access_token": resp.access_token,
-            "device_id": resp.device_id,
-            "user_id": resp.user_id
+                "home_server": home_server,  # e.g. "https://matrix.example.org"
+                "access_token": resp.access_token,  # long cryptographic access token
+                "device_id": resp.device_id,  # device ID, 10 uppercase letters
+                "user_id": resp.user_id  # e.g. "@user:example.org"
             },
             f
         )
 
 
 async def main() -> None:
-    client = AsyncClient("https://matrix.example.org", "@alice:example.org")
 
     # If there are no previously-saved credentials, we'll use the password
     if not os.path.exists(CONFIG_FILE):
-        resp = await client.login("hunter2")
+        print("First time use. Did not find credential file. Asking for homeserver, user, and password to create credential file.")
+        home_server = "https://matrix.example.org"
+        home_server = input(f"Enter your homeserver: [{home_server}] ")
+        if not (home_server.startswith("https://")
+                or home_server.startswith("https://")):
+            home_server = "https://" + home_server
+        user_id = "@user:example.org"
```

Comment:
Lets add a bit of white-space here, it's hard to see whats going on.

Reply:
OK, I can certainly change it, but the pthon linter produced that output. 
If I add whitespaces, or pull the lines together the linter puts it back into this format.
My linter uses pretty much default settings: 
```
autopep8 --in-place --aggressive "$@"
pylama "$@"
```
Since the linter wants it that way, I would respectfully suggest to leave it that way.

Reply:
I don't think there's any particular linting standard throughout the code base, and readability is better than standardisation. Standard linting using `black` might be an idea to consider tho, @poljar?

Reply:
Some files are already formatted with black, some are still waiting on me to go through them and swap the type hints from comments to Python3 type hints. After that we'll set up CI to check with black.

And yeah adding a couple of newlines shouldn't make any linter go wild.